source/common/tcp/conn_pool.h

```diff
@@ -94,6 +94,18 @@ class ActiveTcpClient : public Envoy::ConnectionPool::ActiveClient {
   void onAboveWriteBufferHighWatermark() override { callbacks_->onAboveWriteBufferHighWatermark(); }
   void onBelowWriteBufferLowWatermark() override { callbacks_->onBelowWriteBufferLowWatermark(); }
 
+  // Undo the readDisable done in onEvent(Connected) - now that there is an associated connection,
+  // drain any data.
+  void readEnableIfNew() {
+    if (!associated_before_) {
```

Comment:
What path leads to this being called more than once on a connection?

Reply:
for Envoy, it never is.  AFIK we allow connection reuse and added it for someone who uses it in-house, and I don't want to break things for them.

Reply:
Can you add a comment to that effect?